app/views/auth.py

```diff
@@ -0,0 +1,145 @@
+"""module for user sign up and login"""
+import psycopg2
+from flask import request, jsonify
+from flask_restful import Resource
+from werkzeug.security import generate_password_hash, check_password_hash
+from flask_jwt_extended import create_access_token
+
+from . import cursor
+from app.models.models import User
+from app.utilities import validate_username, validate_email, validate_password
+from app.utilities import check_keys
+from instance.config import conn
+
+
+class UserSignup(Resource):
+    """
+    Class for user signup
+    """
+
+    def post(self):
+        """
+        Add new user
+        """
+        user_details = request.get_json()
+
+        # Check if all required fields have been provided
+        params = ['username', 'email', 'password', 'confirm-password']
+        length = 4
+
+        # Check fields provided
+        if check_keys(user_details, params, length):
+            return check_keys(user_details, params, length)
+
+        # Check if username is valid
+        if validate_username(user_details):
+            return validate_username(user_details)
+
+        # Check if email is valid
+        if validate_email(user_details):
+            return validate_email(user_details)
+
+        # Check password length
+        if validate_password(user_details):
```

Comment:
would be nicer if you wrap this into its own function
```python
def validate_signup()
	if check_keys(user_details, params, length):
            return check_keys(user_details, params, length)
	blah blah...
```

Reply:
Done! Did this for the login view as well.